src/gui/mzroll/projectdockwidget.h

```diff
@@ -31,8 +31,25 @@ class ProjectDockWidget : public QDockWidget
     QColor getSampleColor(mzSample* sample);
     QIcon getSampleIcon(mzSample* sample);
 
+    /**
+     * @brief Map stores the item along with its 
+     * checked and unchecked state.
+     */ 
+    map<QTreeWidgetItem*, bool> saveSampleState; 
+
+    /**
+     * @brief Restores the state of samples as it was before
+     * filtering.
+     */ 
+    void restoreSampleState();
+    
+
 Q_SIGNALS:
     void samplesDeleted();
+    /**
+     * @brief Signal emitted when the item in treeWidget is clicked.
+     */ 
+    void clicked();
```

Comment:
Is this really emitted when the user clicks on anything or when you manually emit it when filters are applied to the tree widget item? If so, then the name should reflect that instead.